driver-core/src/main/java/com/datastax/driver/core/TypeCodec.java

```diff
@@ -548,13 +557,17 @@ public boolean accepts(DataType cqlType) {
      * <p>
      * Implementation notes:
      * <ol>
+     * <li>The default implementation is <em>covariant</em> with respect to the passed
+     * argument (through the usage of {@link TypeToken#isAssignableFrom(TypeToken)}
+     * and <em>it's strongly recommended not to modify this behavior</em>.
+     * This means that, by default, a codec will accept
+     * <em>any subtype</em> of the Java type that it has been created for.</li>
      * <li>The base implementation provided here can only handle non-parameterized types;
      * codecs handling parameterized types, such as collection types, must override
      * this method and perform some sort of "manual"
      * inspection of the actual type parameters.</li>
-     * <li>Similarly, codecs that only accept a partial subset of all possible values,
-     * such as {@link TimeUUIDCodec} or {@link AsciiCodec}, must override
-     * this method and manually inspect the object to check if it
+     * <li>Similarly, codecs that only accept a partial subset of all possible values
```

Comment:
This is actually an oversight from JAVA-965, we shouldn't be mentioning AsciiCodec and TimeUUIDCodec as examples of codecs accepting only certain values, as it is not the case anymore.